storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -310,51 +389,123 @@ func (s *fileSystemSource) loadShardReadersDataIntoShardResult(
 	}
 	if seriesCachePolicy == series.CacheAllMetadata && shardRetriever == nil {
 		s.log.Errorf("shard retriever missing for shard: %d", shard)
-		s.handleErrorsAndUnfulfilled(resultLock, bootstrapResult, shard, tr, shardResult, timesWithErrors)
+		s.handleErrorsAndUnfulfilled(resultLock, runResult, shard, tr,
+			shardResult, timesWithErrors)
 		return
 	}
 
 	for _, r := range readers {
-		if shardResult == nil {
-			resultLock.RLock()
-			results := bootstrapResult.ShardResults()
-			var exists bool
-			shardResult, exists = results[shard]
-			resultLock.RUnlock()
-
-			if !exists {
-				resultLock.Lock()
+		var (
+			timeRange = r.Range()
```

Comment:
hm where does the fs block start -> index block start happen?

Reply:
In `GetOrAddSegment(...)`